src/Microsoft.Azure.WebJobs.Logging/Internal/LogReader.cs

```diff
@@ -155,20 +184,186 @@ public LogReader(CloudTable table)
         {
             TableQuery<RecentPerFuncEntity> rangeQuery = RecentPerFuncEntity.GetRecentFunctionsQuery(queryParams);
 
-            CancellationToken cancellationToken;
-            TableContinuationToken realContinuationToken = Utility.DeserializeToken(continuationToken);
-            var segment = await _instanceTable.SafeExecuteQuerySegmentedAsync<RecentPerFuncEntity>(
-                rangeQuery, 
-                realContinuationToken, 
-                cancellationToken);
+            var iter = new EpochTableIterator(_tableLookup);
+            var results = await iter.SafeExecuteQuerySegmentedAsync<RecentPerFuncEntity>(rangeQuery, queryParams.Start, queryParams.End);
+
+            return results.As<IRecentFunctionEntry>();
+        }
+
+        // Helper to run queries which can span multiple tables.         
+        private class EpochTableIterator
```

Comment:
nit: feel like EpochTableIterator is kind of a misnomer as it's api is not that of an iterator